packages/asset-swapper/contracts/src/NativeOrderSampler.sol

```diff
@@ -0,0 +1,181 @@
+
+    /// @dev Queries the fillable taker asset amounts of native orders.
+    ///      Effectively ignores orders that have empty signatures or
+    ///      maker/taker asset amounts (returning 0).
+    /// @param orders Native orders to query.
+    /// @param orderSignatures Signatures for each respective order in `orders`.
+    /// @param exchange The V3 exchange.
+    /// @return orderFillableTakerAssetAmounts How much taker asset can be filled
+    ///         by each order in `orders`.
+    function getOrderFillableTakerAssetAmounts(
+        LibOrder.Order[] memory orders,
+        bytes[] memory orderSignatures,
+        IExchange exchange
+    )
+        public
+        view
+        returns (uint256[] memory orderFillableTakerAssetAmounts)
+    {
+        orderFillableTakerAssetAmounts = new uint256[](orders.length);
+        for (uint256 i = 0; i != orders.length; i++) {
+            // solhint-disable indent
+            (bool didSucceed, bytes memory resultData) =
+                address(this)
+                    .staticcall
+                    .gas(DEFAULT_CALL_GAS)
+                    (abi.encodeWithSelector(
+                       this.getOrderFillableAmount.selector,
+                       orders[i],
+                       orderSignatures[i],
+                       exchange
+                    ));
+            // solhint-enable indent
+            orderFillableTakerAssetAmounts[i] = didSucceed
+                ? abi.decode(resultData, (uint256))
+                : 0;
+        }
+    }
+
+    /// @dev Queries the fillable taker asset amounts of native orders.
+    ///      Effectively ignores orders that have empty signatures or
+    /// @param orders Native orders to query.
+    /// @param orderSignatures Signatures for each respective order in `orders`.
+    /// @param exchange The V3 exchange.
+    /// @return orderFillableMakerAssetAmounts How much maker asset can be filled
+    ///         by each order in `orders`.
+    function getOrderFillableMakerAssetAmounts(
+        LibOrder.Order[] memory orders,
+        bytes[] memory orderSignatures,
+        IExchange exchange
+    )
+        public
+        view
+        returns (uint256[] memory orderFillableMakerAssetAmounts)
+    {
+        orderFillableMakerAssetAmounts = getOrderFillableTakerAssetAmounts(
+            orders,
+            orderSignatures,
+            exchange
+        );
+        // `orderFillableMakerAssetAmounts` now holds taker asset amounts, so
+        // convert them to maker asset amounts.
+        for (uint256 i = 0; i < orders.length; ++i) {
+            if (orderFillableMakerAssetAmounts[i] != 0) {
+                orderFillableMakerAssetAmounts[i] = LibMath.getPartialAmountCeil(
+                    orderFillableMakerAssetAmounts[i],
+                    orders[i].takerAssetAmount,
+                    orders[i].makerAssetAmount
+                );
+            }
+        }
+    }
+
+    function getOrderFillableAmount(
```

Comment:
Adapted from `DevUtils`. Does not use `compatBalanceOf()` for compatibility with tokens that return extra data.